rotkehlchen/history/events/structures/asset_movement.py

```diff
@@ -0,0 +1,210 @@
+import logging
+from enum import Enum, auto
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.accounting.mixins.event import AccountingEventType
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.location_details import get_formatted_location_name
+from rotkehlchen.exchanges.data_structures import hash_id
+from rotkehlchen.history.events.structures.base import (
+    HISTORY_EVENT_DB_TUPLE_WRITE,
+    HistoryBaseEntry,
+    HistoryBaseEntryType,
+)
+from rotkehlchen.history.events.structures.types import (
+    HistoryEventSubType,
+    HistoryEventType,
+)
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_fval
+from rotkehlchen.types import (
+    Location,
+    TimestampMS,
+)
+from rotkehlchen.utils.misc import ts_ms_to_sec
+
+if TYPE_CHECKING:
+    from more_itertools import peekable
+
+    from rotkehlchen.accounting.mixins.event import AccountingEventMixin
+    from rotkehlchen.accounting.pot import AccountingPot
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class AssetMovementCategory(Enum):
+    """Supported Asset Movement Types so far only deposit and withdrawals"""
+    DEPOSIT = auto()
+    WITHDRAWAL = auto()
+
+
+class AssetMovement(HistoryBaseEntry):
+    """Asset movement event representing deposits and withdrawals on exchanges."""
+
+    def __init__(
+            self,
+            timestamp: TimestampMS,
+            location: Location,
+            category: AssetMovementCategory,
```

Comment:
Again just like previous review I am repeating. Why do you need this? Just pass `event_type: Literal[HistoryEventType.DEPOSIT, HistoryEventType.WITHDRAWAL]` here.

You won't need this articial new ENUM and also save the if checks

Reply:
Yes, using Literal with HistoryEventType is much better. I should have thought of this.

Note that I also added a check for the event type in the deserialization functions, partly because mypy complained about the type now that its restricted with Literal, but also so that it raises an error if somehow it does get the wrong event type during deserialization.